packages/block-editor/src/components/inserter/style.scss

```diff
@@ -166,15 +166,18 @@ $block-inserter-tabs-height: 44px;
 }
 
 .block-editor-inserter__panel-title,
-.block-editor-inserter__panel-title button,
-.components-custom-select-control__menu li {
+.block-editor-inserter__panel-title button {
 	margin: 0 $grid-unit-15 0 0;
 	color: $gray-700;
 	text-transform: uppercase;
 	font-size: 11px;
 	font-weight: 500;
 }
 
+.components-custom-select-control__menu li.is-highlighted {
```

Comment:
Alright, I rebased this and removed the line that causes inheritance of the allcaps text in the custom dropdowns. That actually fixes the margin issue, so you can remove this line.